### PR TITLE
internal/build: fix commit extraction for detached head repo

### PR DESCRIPTION
**Pull request by soar**
If your repository is in "detached HEAD state" when you compile project - commit hash is not detected automatically.

In this repo state `HEAD` file contains just commit hash and not branch name. I propose to use this commit on build stage.

### CODE REVIEW

File: internal/build/env.go [hunk 1]
```diff
@@ -87,6 +88,10 @@ func LocalEnv() Environment {
 	if splits := strings.Split(head, " "); len(splits) == 2 {
 		head = splits[1]
 	} else {
+		commitRe, _ := regexp.Compile("^([0-9a-f]{40})$")
```
Comment (fjl) [inline internal/build/env.go]: Please add a comment to explain the HEAD file format and why this case is needed. The code is pretty subtle already.

Comment (soar) [reply internal/build/env.go]: Added

**Comment by soar** [comment]
@fjl ping

**Comment by soar** [comment]
Up!

**Pull request merged**